[WIP] Add presto webui to presto-proxy

amiorin opened the pull request:
At Zalando, we need to show the webui to our users. We have modified the presto-proxy to use our internal OAuth token. We use Presto without authentication because of this. The new webui with authentication doesn't work for us. My suggestion is to add a webui to presto-proxy with the OAuth flow. It should work with many companies.

My first problem is an error with duplicate-finder-maven-plugin (see screenshot).
<img width="1044" alt="Screenshot 2020-03-25 at 13 12 18" src="https://user-images.githubusercontent.com/32617/77535974-d7ef6c80-6e9b-11ea-82cf-8c8bb4879679.png">

tooptoop4 commented:
similar to https://github.com/lyft/presto/commit/f16770db608d19c19b302b0c5613b6119ee31ea6 ?

amiorin commented:
@tooptoop4 
The technology is similar but we decided to use a different approach. The access_token in the vanilla version of presto-proxy is used for symmetric jwt and we use asymmetric jwt. We modified the presto-proxy to verify and terminate the asymmetric jwt and we modified the jdbc client to use the refresh token saved in the computer of the user after SSO to generate our access token. We also accept the jwt token in the password field for old clients and we implemented the impersonation so that one jwt token is the user while the X-PRESTO-USER is the user executing the query. The authorization for impersonation and execution of SQL statements in Presto cluster is handled by [OPA](https://www.openpolicyagent.org/). The fine-grained authorization at the table level is handled by Hive SQL Standard Based Hive Authorization. We have routing and load balancing but we still need to fix the transaction problem as explained in https://github.com/lyft/presto-gateway/issues/48

<img width="776" alt="Screenshot 2020-03-25 at 20 45 12" src="https://user-images.githubusercontent.com/32617/77578900-a5646480-6ed9-11ea-948a-bc4b406435cd.png">

amiorin commented:
Replaced by #3422
